components/TokenApprovalList/index.tsx

```diff
@@ -0,0 +1,368 @@
+    entries {
+      approved
+      block {
+        timestamp
+      }
+      block_hash
+      data
+      spender_address_hash
+      token_contract_address_hash
+      token_owner_address_hash
+      transaction_hash
+      type
+      udt {
+        id
+        name
+        eth_type
+        icon
+      }
+    }
+    metadata {
+      total_count
+      after
+      before
+    }
+  }
+`
+
+const tokenApprovalListQuery = gql`
+  ${PaginatedTokenApprovalFragment}
+  query tokenApprovalListQuery(
+    $address: HashAddress
+    $after: String
+    $before: String
+    $limit: Int
+    $sorter: [TokenApprovalsSorterInput]
+  ) {
+    erc20_list: token_approvals(
+      input: { address: $address, after: $after, before: $before, limit: $limit, sorter: $sorter, token_type: ERC20 }
+    ) {
+      ...paginateTokenApprovals
+    }
+    erc721_list: token_approvals(
+      input: { address: $address, after: $after, before: $before, limit: $limit, sorter: $sorter, token_type: ERC721 }
+    ) {
+      ...paginateTokenApprovals
+    }
+    erc1155_list: token_approvals(
+      input: { address: $address, after: $after, before: $before, limit: $limit, sorter: $sorter, token_type: ERC1155 }
+    ) {
+      ...paginateTokenApprovals
+    }
+  }
+`
+
+const tokenApprovalListFilteredQuery = gql`
+  ${PaginatedTokenApprovalFragment}
+  query tokenApprovalListFilteredQuery(
+    $address: HashAddress
+    $after: String
+    $before: String
+    $limit: Int
+    $sorter: [TokenApprovalsSorterInput]
+    $token_type: TokenType
+  ) {
+    filtered_list: token_approvals(
+      input: {
+        address: $address
+        after: $after
+        before: $before
+        limit: $limit
+        sorter: $sorter
+        token_type: $token_type
+      }
+    ) {
+      ...paginateTokenApprovals
+    }
+  }
+`
+
+interface Cursor {
+  limit?: number
+  before: string
+  after: string
+}
+
+interface TokenApprovalListVariables extends Nullable<Cursor> {
+  address?: string | null
+  sorter?: GraphQLSchema.TokenApprovalsSorterInput[]
+  token_type?: GraphQLSchema.EthType | null
+}
+
+type TokenApprovalResultList = {
+  entries: Array<TokenApprovalEntryType>
+  metadata: GraphQLSchema.PageMetadata
+}
+
+type TokenApprovalReqResult = {
+  erc20_list: TokenApprovalResultList
+  erc721_list: TokenApprovalResultList
+  erc1155_list: TokenApprovalResultList
+  filtered_list: TokenApprovalResultList
+}
+
+export const fetchTokenApprovalList = (variables: TokenApprovalListVariables) =>
+  client
+    .request<Partial<TokenApprovalReqResult>>(
+      variables.token_type ? tokenApprovalListFilteredQuery : tokenApprovalListQuery,
+      variables,
+    )
+    .then(data => {
+      if (variables.token_type) {
+        return data.filtered_list
+      } else {
+        return {
+          entries: [...data.erc20_list.entries, ...data.erc721_list.entries, ...data.erc1155_list.entries],
+          metadata: data.erc20_list.metadata,
```

Comment:
Why the `entries` is merged from `erc20`, `erc721` and `erc1155` but `metadata` is only from `erc20`

Reply:
Not sure what other ways we can do this, since we have 3 queries for three lists, but only one pagination in UI.

Reply:
As discussed in issue(https://github.com/Magickbase/godwoken_explorer/issues/853#issuecomment-1229125741), all approvals are listed together, so `metadata` is unnecessary because no pagination is be used.